proto/src/main/java/com/google/gson/protobuf/ProtoTypeAdapter.java

```diff
@@ -380,18 +384,23 @@ private EnumValueDescriptor findValueByNameAndExtension(EnumDescriptor desc,
     }
   }
 
-  private static Method getCachedMethod(Class<?> clazz, String methodName)
-      throws NoSuchMethodException {
-    if (!mapOfMapOfMethods.containsKey(methodName)) {
-      mapOfMapOfMethods.put(methodName, new HashMap<Class<?>, Method>());
-    }
+  private static Method getCachedMethod(Class<?> clazz, String methodName,
+      Class<?>... methodParamTypes) throws NoSuchMethodException {
```

Comment:
I take it the method param types don't factor into the key?  Why didn't you need the methodParamTypes before?  Seems a little odd.